docs/src/versioning_releasing.md

```diff
@@ -0,0 +1,162 @@
+
+At this point, you should have a commit on your release branch that you are prepared to release to
+crates.io. Do not merge the pull request yet! Instead, proceed to [release](#releasing-to-cratesio)
+the crates.
+
+### Releasing to crates.io
+
+Releasing a workspace full of interdependent crates can be challenging. Crates must be published in
+the correct order, and any cyclic dependencies that might be introduced via `[dev-dependencies]`
+must be broken. While there is [interest in making this smoother][publish-dev-deps], for now we have
+to muddle through more manually.
+
+[publish-dev-deps]: https://github.com/rust-lang/cargo/issues/4242
+
+1. Authenticate with `cargo login` using a Github account with the appropriate access to the Lucet
+   repository. You should only have to do this once per development environment.
+
+1. Ensure that you have the commit checked out that you would like to release.
+
+1. Ensure that the version in all of the Lucet `Cargo.toml` files matches the version you expect to
+   release. Between releases, the versions will end in `-dev`; if this is still the case, you'll
+   need to replace this version with the appropriate version according to the guidelines above,
+   likely through a PR.
+
+1. Edit `lucet-validate/Cargo.toml` and make the following change (note the leading `#`):
+
+   ```diff
+    [dev-dependencies]
+   -lucet-wasi-sdk = { path = "../lucet-wasi-sdk", version = "=0.5.2" }
+   +#lucet-wasi-sdk = { path = "../lucet-wasi-sdk", version = "=0.5.2" }
+    tempfile = "3.0"
+   ```
+   
+   This breaks the only cycle that exists among the crates as of `0.5.1`; if other cycles develop,
+   you'll need to similarly break them by temporarily removing the dev dependency.
+
+1. Begin publishing the crates in a topological order by `cd`ing to the each crate and running
+   `cargo publish --allow-dirty` (the tree should only be dirty due to the cycles broken
+   above). While we would like to run `cargo publish --dry-run` beforehand to ensure all
+   of the crates will be successfully published, this will fail for any crates that depend on other
+   Lucet crates, as the new versions will not yet be available to download.
+
+   Do not worry too much about calculating the order ahead of time; if you get it wrong, `cargo
+   publish` will tell you which crates need to be published before the one you tried. An order which
+   worked for the `0.5.1` release was:
+   
+   1. `lucet-module`
+   1. `lucet-validate`
+   1. `lucetc`
+   1. `lucet-wasi-sdk`
+   1. `lucet-objdump`
+   1. `lucet-runtime-macros`
+   1. `lucet-runtime-internals`
+   1. `lucet-runtime-tests`
+   1. `lucet-runtime`
+   1. `lucet-wasi`
+
+   It is unlikely but not impossible that a publish will fail in the middle of this process, leaving
+   some of the crates published but not others. What to do next will depend on the situation; please
+   consult with the Lucet team.
+
+Congratulations, the new crates are now on crates.io! 🎉
```

Comment:
I'm not familiar with publishing crates on crates.io, so please forgive this naive question.  Would it make sense to add some kind of sanity check step to make sure the lucet crates that were just published are actually available at crates.io?  I don't want to break other people's toys.

Reply:
Added a step for this, thanks!